src/esperanza/finalizationstate.cpp

```diff
@@ -187,19 +196,25 @@ void FinalizationState::IncrementDynasty() {
 }
 
 ufp64::ufp64_t FinalizationState::GetCollectiveRewardFactor() {
-  uint32_t epoch = m_current_epoch;
   bool isLive = GetEpochsSinceFinalization() <= 2;
 
   if (!DepositExists() || !isLive) {
     return 0;
   }
 
+  // we use "m_current_epoch - 2" epoch to retrieve dynasty votes because:
+  // -1 as m_current_epoch is already incremented inside of InitializeEpoch
+  // before this function is invoked
+  // -1 because when we store votes we use vote.m_target_epoch which is always
+  // one below the m_current_epoch
+  const uint32_t checkpoint_epoch = m_current_epoch - 2;
```

Comment:
This name I believe is quite confusing. What we have here is the last finalized epoch since `isLive` is true, this cannot be otherwise.

Reply:
changed to `m_last_finalized_epoch` and added assert that `m_last_finalized_epoch == m_current_epoch - 2`